### PR TITLE
Feature/276 display last visited course

### LINKED ISSUE #276: List the last visited course at first in the "my courses" view
# User Story

### As a: 

Student

### I want:

to see the course tile first, that I last visited

### so that:

I can directly jump right into where I left off.

## Additional info:
after discussing with david both categories are sorted by bottles in the teacher's section

### PR DESCRIPTION
**Pull request by jsinisa**
## Description:
Sorted output of last visited courses on your dashboard

Closes #276 

## Improvements
You can see your last attended courses and can directly jump right into where you left off

## Known Issues:
Requires algorithm to save the courses by last visited order
1. How do you recognize the last course you attended? (timestamp?)
2.After saving how the courses are spent on the dashboard in the correct order
------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: api/src/models/User.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import * as crypto from 'crypto';
 import {isNullOrUndefined} from 'util';
 import {isEmail} from 'validator';
 import * as errorCodes from '../config/errorCodes'
+import {ICourse} from '../../../shared/models/ICourse';
```
Comment (felixbrucker) [inline api/src/models/User.ts]: This should be removed as its not used anywhere

File: app/webFrontend/src/app/app.module.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ import {
   DownloadFileService,
   FreeTextUnitService,
   LectureService,
+  TaskService,
```
Comment (felixbrucker) [inline app/webFrontend/src/app/app.module.ts]: This should be removed as it's a merge error and got removed upstream

File: app/webFrontend/src/app/course/course.component.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ import {ShowProgressService} from '../shared/services/show-progress.service';
 
 
 
+
```
Comment (felixbrucker) [inline app/webFrontend/src/app/course/course.component.ts]: Please remove this unnecessary blank line

File: app/webFrontend/src/app/shared/services/data.service.ts [hunk 4]
```diff
@@ -9,6 +9,7 @@ import {IUser} from '../../../../../../shared/models/IUser';
 import {ICourse} from '../../../../../../shared/models/ICourse';
 import {IDownload} from '../../../../../../shared/models/IDownload';
 
+
```
Comment (felixbrucker) [inline app/webFrontend/src/app/shared/services/data.service.ts]: Please remove this unnecessary blank line

File: app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts [hunk 5]
```diff
@@ -2,13 +2,15 @@ import {Component} from '@angular/core';
...
 import {DashboardBaseComponent} from '../dashboard-base-component';
+import {SortUtil} from '../../../shared/utils/SortUtil';
 
 
 @Component({
   selector: 'app-dashboard-student',
   templateUrl: './dashboard-student.component.html',
   styleUrls: ['./dashboard-student.component.scss']
 })
+
```
Comment (felixbrucker) [inline app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts]: Please remove this unnecessary blank line

File: app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts [hunk 6]
```diff
@@ -19,16 +21,20 @@ export class DashboardStudentComponent extends DashboardBaseComponent {
   }
 
   ngOnInit() {
+
```
Comment (felixbrucker) [inline app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts]: Please remove this unnecessary blank line

File: shared/models/IUser.ts [hunk 7]
```diff
@@ -1,4 +1,6 @@
 import {IFile} from './IFile';
+import {ICourse} from './ICourse';
```
Comment (felixbrucker) [inline shared/models/IUser.ts]: This should be removed as its not used anywhere

Comment (felixbrucker) [reply shared/models/IUser.ts]: there is an added empty line below which should be removed as well ;)

File: app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts [hunk 8]
```diff
@@ -19,15 +20,17 @@ export class DashboardStudentComponent extends DashboardBaseComponent {
...
 
   ngOnChanges() {
     this.sortCourses();
   }
 
-  sortCourses() {
+  async sortCourses() {
     this.myCourses = [];
     this.availableCourses = [];
+    SortUtil.sortByLastVisitedCourses(this.allCourses, this.userService.user.lastVisitedCourses);
```
Comment (felixbrucker) [inline app/webFrontend/src/app/start/dashboard/dashboard-student/dashboard-student.component.ts]: Please keep this space here

File: shared/models/IUser.ts [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import {IFile} from './IFile';
+
```
Comment (felixbrucker) [inline shared/models/IUser.ts]: Please remove this empty line

**Comment by jsinisa** [comment]
after discussing with david both categories are sorted by allCourses in the teacher's section

**Comment by aeimer** [comment]
Tested locally. Works.

**Pull request merged**